odoo/tools/misc.py

```diff
@@ -1176,6 +1176,25 @@ def formatLang(env, value, digits=None, grouping=True, monetary=False, dp=False,
             res = '%s %s' % (currency_obj.symbol, res)
     return res
 
+def parse_date(env, value, lang_code=False):
```

Comment:
move it below `format_date` as in upper versions, this way we avoid having twice the same code when forward ported